omnibus/files/private-chef-cookbooks/private-chef/libraries/ec_postgres.rb

```diff
@@ -3,11 +3,28 @@ class EcPostgres
   def self.with_connection(node, database = 'template1', opts = {})
     require 'pg'
     postgres = node['private_chef']['postgresql'].merge(opts)
-    connection = ::PGconn.open('user' => postgres['db_superuser'],
-                               'host' => postgres['vip'],
-                               'password' => postgres['db_superuser_password'],
-                               'port' => postgres['port'],
-                               'dbname' => database)
+    connection = nil
+    retries = 5
+    begin
+      connection = ::PGconn.open('user' => postgres['db_superuser'],
+                                 'host' => postgres['vip'],
+                                 'password' => postgres['db_superuser_password'],
+                                 'port' => postgres['port'],
+                                 'dbname' => database)
+    rescue e
```

Comment:
Yeah, pg_conn shoves everything into PGError. In postgres preflight validator I had to match on postgres error strings :/